lib/govuk_markdown/renderer.rb

```diff
@@ -81,7 +81,8 @@ def hrule
     end
 
     def preprocess(document)
-      Preprocessor.new(document).inject_inset_text
+      rendered_doc = Preprocessor.new(document).inject_inset_text
+      Preprocessor.new(rendered_doc).inject_details
```

Comment:
I think this might be nicer if it was chained:

```
Preprocessor
  .new(document)
  .inject_inset_text
  .inject_details
  .document
```

You'd need to change the `#gsub` calls in the inject methods to the `!` variant and have the method return `self`.

Perhaps there's a better name for `attr_reader :document` too, as we're passing a document in. Maybe `buffer` adn then add a `#to_s` method? 🤔

Reply:
great suggestions! Have implemented your comments - I've gone with `output` instead of `buffer`, maybe not the best name but was more understandable to me